cleaning_data_review.py

```diff
@@ -0,0 +1,156 @@
+############################################################
+  
+file_contents_noguthenberg = delete_guthenberg_info(file_contents)
+#print(file_contents_noguthenberg)
+
+
+"""save as file"""
+"""#Combine the contents into a single string with a separator
+combined_content = "\n\n---\n\n".join(file_contents.values())
+# Save the combined content to a single .txt file
+output_file_path = 'combined_texts4.txt'
+with open(output_file_path, 'w', encoding='utf-8') as f:
+    f.write(combined_content)"""
+ 
+combined_content = ""
+for filename, content in file_contents.items():
+    combined_content += f"\n\n--- Start of {filename} ---\n\n{content}\n\n--- End of {filename} ---\n\n"
+
+# Save combined content
+combined_output_path = 'combined_texts2.txt'
+with open(combined_output_path, 'w', encoding='utf-8') as f:
+    f.write(combined_content)
+
+#open folder
+with open('combined_texts2.txt', 'r', encoding='utf-8') as f:
+    # Read the entire file
+    content = f.read()
+
+#####################################
+# Removing paragraph with specific pattern
+# Remove article starting with the specific sentence
+start_sentence = "Copyright pages exist to tell you"
+content = remove_paragraph_starting_with(content, start_sentence)
+
+start_sentence = "This ebook is"
+content = remove_paragraph_starting_with(content, start_sentence)
+#print(content)
+output_file_path = 'combinedtexts2_withoutebooks'
+with open(output_file_path, 'w', encoding='utf-8') as f:
+    f.write(content)
+
+############################################################
+
+# Read the content from the file
+with open('combinedtexts_handclean.txt', 'r', encoding='utf-8') as f: 
+    newcontent = f.read()
+
+# Define the start sentence regular expression
+start_sentence_pattern = r'\[(F|f)ootnote \d+:.*?\]'
+
+# Remove the paragraph that starts with the specified sentence
+content = remove_paragraph_starting_with(newcontent, start_sentence_pattern)
+
+# Print the cleaned content
+#print(content)
+
+######################################################################################
+keyword = "footnote"
+
+# Remove all paragraphs that contain the specified keyword
+cleaned_content = remove_paragraph_including(content, keyword)
+
+# Print the cleaned content or save to a new file if it's large
+#print(cleaned_content)
+
+output_file_path = 'combinedtexts2_handclean_removedfootnotes'
+with open(output_file_path, 'w', encoding='utf-8') as f:
+    f.write(cleaned_content)
+
+# Read the UTF-8 with BOM file and write it as normal UTF-8
+input_file_path = 'data\The Complete Works of William Shakespeare.txt'
+output_file_path = 'Shakespear_utf8.txt'
+######################################################
+print(f"Combined content with filename separators has been written to {output_file_path}")  
+convert_to_UTF8 (input_file_path,output_file_path)
+
+######################################################
+roman_numeral_pattern = r'\b[MDCLXVI]+\b'    # Roman numerals

```

Comment:
all these constant variable should be after imports in top of the file. They can be upper case because they are not to be changed
e.g.
```
RE_ROMAN_NUMERAL = r'\b[MDCLXVI]+\b' 
RE_ARABIC_NUMERAL = '\b\d+\b'
RE_CHARACTER_NAME = r'\b[A-Z]{1,}\b'  #  or first word in the sentence?
```
```